### PR TITLE
gh-99184: Bypass instance attribute access in `repr` of `weakref.ref`

### PR DESCRIPTION
**Pull request by sobolevn**
Consider this as a fix proposal :)

Thanks a lot to @MojoVampire for the idea.

* Issue: gh-99184

### CODE REVIEW

File: Objects/weakrefobject.c [hunk 1]
```diff
@@ -170,10 +170,7 @@ weakref_repr(PyWeakReference *self)
     }
 
     Py_INCREF(obj);
```
Comment (sobolevn) [inline Objects/weakrefobject.c:172]: Do we still need it? 🤔

Comment (sobolevn) [reply Objects/weakrefobject.c:172]: Yes, without it:

```
Fatal Python error: _PyObject_AssertFailed: _PyObject_AssertFailed
Python runtime state: initialized

Current thread 0x00000001039365c0 (most recent call first):
  File "/Users/sobolev/Desktop/cpython/Lib/test/test_weakref.py", line 116 in test_basic_ref
  File "/Users/sobolev/Desktop/cpython/Lib/unittest/case.py", line 579 in _callTestMethod
```

**Comment by sobolevn** [comment]
You were suggested reviewers :)

**Comment by sobolevn** [comment]
Note for revierers about dynamic `__weakref__`. One might think that this PR can break existing dynamic `__weakref__` assignment. For example:

```python
>>> import weakref
>>> class A: ...
... 
>>> a = A()
>>> w = weakref.ref(a)
>>> a1 = A()
>>> a1.__weakref__
>>> a1.__weakref__ = w
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
AttributeError: attribute '__weakref__' of 'A' objects is not writable
```

But, as you can see it is not writable. So, it should not affect anything really. 
I am going to update the branch and trigget all buildbots, just to be sure.

**Review by carljm — APPROVED** [review]
LGTM

**Pull request merged**